src/API/Notes.php

```diff
@@ -415,6 +415,29 @@ public function prepare_query_for_response( $query ) {
 		return admin_url( $query );
 	}
 
+	/**
+	 * Maybe add a nonce to a URL.
+	 *
+	 * @link https://codex.wordpress.org/WordPress_Nonces
+	 *
+	 * @param string $url The URL needing a nonce.
+	 * @param string $action The nonce action.
+	 * @param string $name The nonce anme.
+	 * @return string A fully formed URL.
+	 */
+	private function maybe_add_nonce_to_url( string $url, string $action = '', string $name = '' ) : string {
+		if ( empty( $action ) ) {
+			return $url;
+		}
+
+		if ( empty( $name ) ) {
+			// Allow wp_nonce_url to use a default paramater name.
+			$name = null;
+		}
+
+		return wp_nonce_url( $url, $action, $name );
```

Comment:
One thing to note here is that `wp_nonce_url` escapes `&` and could cause issues with broken links. For example if I passed in `https://site.com/?do_update=true&_nonce=update` it would end up `https://site.com/?do_update=true&amp;amp;_nonce=update` which is broken.

Perhaps use `add_query_arg( $name, wp_create_nonce( $action ), $url );`

Reply:
Thanks, nice catch! I followed your suggestion in cb3c71a.